ui/components/component-library/button-base/button-base.js

```diff
@@ -195,7 +195,7 @@ ButtonBase.propTypes = {
   /**
    * textProps accepts all the props from Icon
    */
-  textProps: PropTypes.shape(Text.propTypes),
+  textProps: PropTypes.object,
   /**
    * ButtonBase accepts all the props from Box
    */
```

Comment:
## Buttons
Before/After screencast showing no visual regressions for all button components

### Before

https://github.com/MetaMask/metamask-extension/assets/8112138/b09f8ab0-75eb-4d87-a6c2-4e8a08a1df04

### After

https://github.com/MetaMask/metamask-extension/assets/8112138/0b62e67c-379b-41a0-9ef2-98aa59bb3e9f